neo/IO/Helper.cs

```diff
@@ -131,7 +131,14 @@ public static T[] ReadSerializableArray<T>(this BinaryReader reader, int max = 0
 
         public static byte[] ReadVarBytes(this BinaryReader reader, int max = 0x1000000)
         {
-            return reader.ReadBytes((int)reader.ReadVarInt((ulong)max));
+            max = (int)reader.ReadVarInt((ulong)max);
+
+            if (reader.BaseStream.CanSeek && (reader.BaseStream.Length - reader.BaseStream.Position) < max)
```

Comment:
If `reader.BaseStream.CanSeek == false`, it is the same as the original implemention. We should limit `max`.

Reply:
I checked everywhere, the maximum `max` is 16MB, no problem.

Reply:
So the fix should be in neo-vm.

Reply:
https://github.com/neo-project/neo/blob/f454e986a01aa979af03f89afa4530d3e279fb76/neo/SmartContract/ApplicationEngine.cs#L125-L132

The stack element has been limited to 1MB here.

Reply:
You are right